web-ui/src/main/resources/catalog/components/search/facets/FacetDimensionDirective.js

```diff
@@ -220,10 +235,41 @@
                 var initialMaxItems = 5;
                 scope.initialMaxItems = initialMaxItems;
                 scope.maxItems = initialMaxItems;
-                scope.toggleAll = function() {
-                  scope.maxItems = (scope.maxItems == Infinity) ?
-                      initialMaxItems : Infinity;
-                };
+
+                if (scope.category) {
+                  scope.evalPageSize = function() {
+                    scope.morePageSize = Math.min (scope.category.length - scope.maxItems, scope.pageSize);
+                    scope.minusPageSize = Math.min (scope.maxItems - initialMaxItems, scope.pageSize);
+                  }
```

Comment:
You could achieve this in a simpler way by simply defining `scope.getMorePageSize()` and `scope.getLessPageSize()` functions, and then use these in the HTML template instead of variables.